source/Plugins/SteamLibrary/SteamLibrary.cs

```diff
@@ -125,6 +127,73 @@ internal List<Game> GetInstalledGamesFromFolder(string path)
             return games;
         }
 
+        internal List<Game> GetInstalledGoldSrcModsFromFolder(string path)
+        {
+            var games = new List<Game>();
+
+            var firstPartyMods = new string[] { "bshift", "cstrike", "czero", "czeror", "dmc", "dod", "gearbox", "ricochet", "tfc", "valve"};
+
+            var dirInfo = new DirectoryInfo(path);
+
+            foreach (var folder in dirInfo.GetDirectories().Where(a => Array.IndexOf(firstPartyMods, a.Name) == -1).Select(a => a.FullName))
+            {
+                try
+                {
+                    var game = GetInstalledModFromFolder(folder, ModInfo.ModType.HL);
+                    games.Add(game);
+                }
+                catch (Exception exc)
+                {
+                    // gameinfo.txt may not exist or may be invalid
+                    logger.Error(exc, $"Failed to get information about installed GoldSrc mod from: {path}");
+                }
+            }
+
+            return games;
+        }
+
+        internal List<Game> GetInstalledSourceModsFromFolder(string path)
+        {
+            var games = new List<Game>();
+
+            foreach (var folder in Directory.GetDirectories(path))
+            {
+                try
+                {
+                    var game = GetInstalledModFromFolder(folder, ModInfo.ModType.HL2);
+                    games.Add(game);
+                }
+                catch (Exception exc)
+                {
+                    // gameinfo.txt may not exist or may be invalid
+                    logger.Error(exc, $"Failed to get information about installed Source mod from: {path}");
+                }
+            }
+
+            return games;
+        }
+
+        static internal Game GetInstalledModFromFolder(string path, ModInfo.ModType modType)

```

Comment:
Remove static and inject complete` SteamLibrary` into `SteamMetadataProvider` (you can replace `Settings` object since it comes from SteamLibrary`), that way you can also remove `Id` workaround on line 488.